Rubberduck.CodeAnalysis/Inspections/Concrete/EmptyMethodInspection.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Common;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.Parsing.VBA;
+using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Resources.Experimentals;
+using static Rubberduck.Parsing.Grammar.VBAParser;
+
+namespace Rubberduck.Inspections.Concrete
+{
+
+    /// <summary>
+    /// Identifies empty module member blocks.
+    /// </summary>
+    /// <why>
+    /// Methods containing no executable statements are misleading as they appear to be doing something which they actually don't.
+    /// This might be the result of delaying the actual implementation for a later stage of development, and then forgetting all about that.
+    /// </why>
+    /// <example hasResults="true">
+    /// <![CDATA[
+    /// Sub Foo()
+    ///     ' ...
+    /// End Sub
+    /// ]]>
+    /// </example>
+    /// <example hasResults="false">
+    /// <![CDATA[
+    /// Sub Foo()
+    ///     MsgBox "?"
+    /// End Sub
+    /// ]]>
+    /// </example>
+    [Experimental(nameof(ExperimentalNames.EmptyBlockInspections))]
+    internal class EmptyMethodInspection : InspectionBase// EmptyBlockInspectionBase
+    {
+        public EmptyMethodInspection(RubberduckParserState state)
+            : base(state) { }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            // Exclude empty members in user interfaces, as long as all members of the interface are empty,
+            // since some VB users might use concrete user defined classes as interfaces,
+            // while RD marks them as interfaces all the same.
+
+            return UserDeclarations.OfType<ModuleBodyElementDeclaration>()
```

Comment:
Can we use `State.DeclarationFinder.UserDeclarations(DeclarationType.Member)` here instead of `UserDeclarations`? The difference is that this will use a dictionary lookup to only get module body elements instead of iterating over all declarations and testing for the type afterwards.

Reply:
Yes, we could, by implementing same functionality from `QualifiedContext`:
`State.DeclarationFinder.UserDeclarations(DeclarationType.Member)
.Where(member => !member.IsIgnoringInspectionResultFor(...))` since `UserDeclarations` is doing this already.
Should I go for it?

Reply:
I was shortly wondering whether it might be more efficient to filter for ignored declarations later, but after reviewing the corresponding code, I think the check is a rather fast operation and can be done immediately.

So, you can implement it as proposed or do the ignored check at the end. I have no real preference there.

Reply:
OK, added the extensions and now using `DeclarationFinder`, as requested.